pulp_rpm/app/serializers/modulemd.py

```diff
@@ -57,6 +48,20 @@ class ModulemdSerializer(NoArtifactContentSerializer):
     profiles = serializers.JSONField(help_text=_("Modulemd profiles."), allow_null=True)
     snippet = serializers.CharField(help_text=_("Modulemd snippet"), write_only=True)
 
+    def create(self, validated_data):
+        """
+        Create and return a new `Modulemd` instance, given the validated data.
+        """
+        snippet = validated_data["snippet"]
+        validated_data["digest"] = hashlib.sha256(snippet.encode()).hexdigest()
+        # In django, we can't add items to a m2m relationship on the main object instantiation.
+        # First we need to create the main object, then add things to the m2m field.
+        # https://stackoverflow.com/a/50015229
+        packages = validated_data.pop("packages")
+        modulemd = super().create(validated_data)
+        modulemd.packages.set(packages)
+        return modulemd
```

Comment:
My only concern is that we have no way to go and mark packages as modular.  If you upload a package, it's non-modular.  If you then create a module that includes that package, nothing can change that (and we wouldn't want to either, it might break other repos)

I don't see a good way of solving that problem unless we ditch `is_modular` entirely and exclusively rely on the modules present in the repo to decide which packages are or are not modular, which is similar to how we work with errata and comps.

I can't remember if there's any specific reasons we can't do that which informed the current approach, though.  @ggainey @ipanova can you think of any?

Of course, migrating away from the current approach would be very painful

Reply:
I don't remember why we decided to go that path in first place and I do not see why we could not switch to look whether a packages is present in a module.

Moreover, it might be a better approach of not having the flag on the package. As you mentioned we would need to flip it back and forth when uploading/syncing/removing module.

Reply:
I don't want to block this PR on this basis FYI, it's just a further bug to deal with